core/evm.go

```diff
@@ -18,12 +18,33 @@ package core
 
 import (
 	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/consensus"
 	"github.com/celo-org/celo-blockchain/core/types"
 	"github.com/celo-org/celo-blockchain/core/vm"
 	"github.com/celo-org/celo-blockchain/core/vm/vmcontext"
+	"github.com/celo-org/celo-blockchain/params"
 )
 
+// ChainContext supports retrieving chain data and consensus parameters
+// from the blockchain to be used during transaction processing.
+type ChainContext interface {
+	// Engine retrieves the blockchain's consensus engine.
+	Engine() consensus.Engine
+
+	// GetHeader returns the hash corresponding to the given hash and number.
+	GetHeader(common.Hash, uint64) *types.Header
+
+	// GetHeaderByNumber returns the hash corresponding number.
+	// FIXME: Use of this function, as implemented, in the EVM context produces undefined behavior
+	// in the pressence of forks. A new method needs to be created to retrieve a header by number
```

Comment:
```suggestion
	// in the presence of forks. A new method needs to be created to retrieve a header by number
```

Though, since we will be disabling reorgs, maybe we don't need this FIXME at all.